### PR TITLE
Refactored app to use .apply.opts

### PR DESCRIPTION
**Pull request by philippjfr**
Ensures that all the majority of widget updates can happen with re-rendering.

### CODE REVIEW

File: app.ipynb [hunk 1]
```diff
@@ -388,6 +388,9 @@
     "        self.graph_pane.append(\n",
     "            layout\n",
     "        )\n",
+    "        \n",
+    "    def __repr__(self):\n",
+    "        return self.name\n",
```
Comment (philippjfr) [inline app.ipynb:393]: Ooops, feel free to delete this.

Comment (WesleyTheGeolien) [reply app.ipynb:393]: Thanks for this @philippjfr

File: app.ipynb [hunk 2]
```diff
@@ -57,14 +57,14 @@
     "    # Percentage => Base value + percentage\n",
     "    calculation_type = pn.widgets.RadioButtonGroup(options=['Absolute', 'Relatif', 'Percentage'], align='end')\n",
     "    # Replacement value\n",
-    "    spinner = pn.widgets.Spinner(name='Replacement Value', value=0, align='start')\n",
+    "    spinner = pn.widgets.IntInput(name='Replacement Value', value=0, align='start')\n",
```
Comment (WesleyTheGeolien) [inline app.ipynb:60]: We need to pin a minimum version of panel to 0.10 to ensure this works correctly for further use

File: app.ipynb [hunk 3]
```diff
@@ -270,40 +269,67 @@
...
+    "\n",
+    "    def _color_levels(self):\n",
+    "        if self.colormap_delta.value <= 0:\n",
+    "            return None\n",
+    "        return list(range(self.colormap_min.value, self.colormap_max.value, self.colormap_delta.value)) + [self.colormap_max.value]\n",
+    "            \n",
+    "    def _colorbar_opts(self):\n",
+    "        if self.colormap_delta.value <= 0:\n",
+    "            return {}\n",
+    "        ticks = self._levels()\n",
```
Comment (WesleyTheGeolien) [inline app.ipynb:287]: pretty sure `self._levels` should be `self._color_levels`

File: app.ipynb [hunk 4]
```diff
@@ -270,40 +269,67 @@
...
+    "        if len(levels) > 8:\n",
```
Comment (WesleyTheGeolien) [inline app.ipynb:288]: change levels to ticks

**Review by WesleyTheGeolien — APPROVED** [review]

**Pull request merged**